docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.rst

```diff
@@ -0,0 +1,63 @@
+Enable OAuth2/OIDC support for Apache Kafka® REST proxy
+========================================================
+
+Secure your Apache Kafka® resources by integrating OAuth 2.0/OpenID Connect (OIDC) with the Karapace REST proxy and enabling REST proxy authorization. This setup ensures that only authorized individuals can manage Apache Kafka resources through both token-based authentication and access control rules.
+
+OAuth2/OIDC token handling
+---------------------------
+
+Karapace processes the JSON Web Token (JWT) obtained from the Authorization HTTP header, specifically when employing the Bearer authentication scheme. This allows OAuth2/OIDC credentials to be supplied directly to the REST proxy, which uses the provided token to authorize requests to Apache Kafka. When a Bearer token is presented, Kafka clients configured by Karapace use the SASL OAUTHBEARER mechanism to send the JWT for validation.
+
+
+Authorization enforcement
+----------------------------
+
+The underlying Aiven for Apache Kafka® service uses the ``sub`` claim from the JWT as the username, checking it against configured Access Control Lists (ACLs) to authorize operations on Kafka resources.
```

Comment:
A small thing here, the default is indeed the `sub` claim, however on the Kafka side this is configurable via `kafka.sasl_oauthbearer_sub_claim_name`, as also mentioned in https://docs.aiven.io/docs/products/kafka/howto/enable-oidc#enable-oauth2-oidc-via-aiven-console

Reply:
And currently the service user matching the value in the claim is needed, just the ACL entry is not enough.

Reply:
Thanks both for your feedback. I have made some changes to the text and would appreciate it if you could review it again.